electrumsv/wallet.py

```diff
@@ -3921,10 +4022,27 @@ def start(self, network: Optional[Network]) -> None:
         for account in self.get_accounts():
             account.start(network)
 
+        main_server: Optional[SelectionCandidate] = None
         if self._network is not None:
-            # TODO(1.4.0) Networking. Every wallet has connections to reference servers. Is this
-            #     the right place to connect?
+            server_url = self._storage.get('main_server', "")
+            if server_url == "":
+                # NOTE: Defaults to whatever the first server in the list is
+                main_server = self._network.get_api_servers_for_headers()[0]
+                assert main_server is not None
+                assert main_server.api_server is not None
+                self._storage.put('main_server', main_server.api_server.url)
+            else:
+                for server in self._network.get_api_servers_for_headers():
+                    assert server.api_server is not None
+                    if server.api_server.url == server_url:
+                        main_server = server
+
+            assert main_server is not None
+            app_state.async_.spawn_and_wait(
+                self._set_main_server, main_server, SwitchReason.user_set)
             self._setup_server_connections()
+            self._worker_startup_reorg_check = app_state.async_.spawn(
```

Comment:
This is already only being launched in online mode AFAICT, so it should not check for offline inside the task code.